### PR TITLE
Idan/265 donor add to calendar

### PR DESCRIPTION
**Pull request by idanlevi1**

### CODE REVIEW

File: packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx [hunk 1]
```diff
@@ -15,19 +15,75 @@ import Cancellation from "../../assets/images/cancelation.svg";
 import Whatsapp from "../../assets/images/whatsup-color-big.svg";
 import TrashIcon from "../../assets/icons/trash.svg";
 import UpcomingDonationInfo from "./UpcomingDonationInfo";
+import makeUrls, { TCalendarEvent } from "add-event-to-calendar";
+import ICalendarLink from "react-icalendar-link";
+import Menu from "@material-ui/core/Menu";
```
Comment (yaron1m) [inline packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx:20]: `Donor` is not dependent on Material UI, so you can't import it here.

Please ask Merav or Matar how this should look in the client, and let's implement this without using Material UI.

Could we use a popup for this?

File: packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx [hunk 2]
```diff
@@ -61,11 +117,37 @@ export default function UpcomingDonationScreen({
 
               <div className={styles.detailLabel}>מתי?</div>
               <div className={styles.detailValue}>
-                {donationDate.toLocaleDateString(
+                {eventDateStart.toLocaleDateString(
                   "he-He",
                   DateUtils.ShortDateFormat
                 )}
               </div>
+              <div className={styles.link} onClick={handleClick}>
```
Comment (yaron1m) [inline packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx:125]: Please extract the entire code of the `Add to Calendar` **button** to a dedicated component

File: packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx [hunk 3]
```diff
@@ -15,19 +15,75 @@ import Cancellation from "../../assets/images/cancelation.svg";
...
   onCancel,
   bookedAppointment,
 }: UpcomingDonationScreenProps) {
-  const donationDate = new Date(bookedAppointment.donationStartTimeMillis);
+  const eventDateStart = new Date(bookedAppointment.donationStartTimeMillis);
+  const eventDateEnd = new Date(eventDateStart.getTime() + 90 * 60000);
+  const eventLocation = `בית החולים ${LocaleUtils.getHospitalName(
+    bookedAppointment.hospital
+  )}`;
+  const eventDescription = `זכרון מנחם - ${eventLocation} - ${eventDateStart.toLocaleDateString(
```
Comment (yaron1m) [inline packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx:49]: Please use DateUtils in `common`

File: packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx [hunk 4]
```diff
@@ -15,19 +15,75 @@ import Cancellation from "../../assets/images/cancelation.svg";
...
+];
+
 export default function UpcomingDonationScreen({
   fullName,
   onCancel,
   bookedAppointment,
 }: UpcomingDonationScreenProps) {
-  const donationDate = new Date(bookedAppointment.donationStartTimeMillis);
+  const eventDateStart = new Date(bookedAppointment.donationStartTimeMillis);
+  const eventDateEnd = new Date(eventDateStart.getTime() + 90 * 60000);
```
Comment (yaron1m) [inline packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx:45]: appointment calendar length should be a constant (i.e. a dedicated variable to be changed easily)

File: packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx [hunk 5]
```diff
@@ -15,19 +15,75 @@ import Cancellation from "../../assets/images/cancelation.svg";
...
+
+  const handleClick = () => {
+    setAnchorEl(true);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(false);
+  };
+
+  const addToCalendarByType = (calendar: CalendarType) => {
```
Comment (yaron1m) [inline packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx:80]: Please create a dedicated Util in `common` that accepts a file type and returns a link to an invite.

This should be in common so we could use it in the backend as well (when sending emails).

Also, please add tests to it

**Review by yaron1m — COMMENTED** [review]
Please add screenshots (of the screen and the events)

File: packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx (hunk 1 above)
Comment (idanlevi1) [reply packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx]: Ok, I removed Material UI usage

**Comment by yaron1m** [comment]
Hi, there are still some unresolved comments.

Please let me know when to review this again

**Pull request closed without merging**